contracts/eos/BancorConverter/BancorConverter.cpp

```diff
@@ -173,34 +173,21 @@ ACTION BancorConverter::setreserve(symbol_code converter_currency_code, symbol c
     const auto& converter = converters_table.get(converter_currency_code.raw(), "converter does not exist");
     require_auth(converter.owner);
     
-    settings settings_table(get_self(), get_self().value);
-    const auto& st = settings_table.get("settings"_n.value, "settings do not exist");
-
-    string error = string("ratio must be between 1 and ") + std::to_string(MAX_RATIO);
+    const string error = string("ratio must be between 1 and ") + std::to_string(MAX_RATIO);
     check(ratio > 0 && ratio <= MAX_RATIO, error.c_str());
     
     check(is_account(contract), "token's contract is not an account");
     check(currency.is_valid(), "invalid reserve symbol");
     
     reserves reserves_table(get_self(), converter_currency_code.raw());
-    auto reserve = reserves_table.find(currency.code().raw());
-    asset smart_supply = get_supply(st.multi_token, converter.currency.code());
-
-    if (reserve != reserves_table.end()) {
-        check(reserve->contract == contract, "cannot update the reserve contract name");
-        reserves_table.modify(reserve, get_self(), [&](auto& r) {
-            r.ratio = ratio;
-        });
-        auto reserve_balance = reserve->balance.amount / pow(10, currency.precision()); 
-        EMIT_PRICE_DATA_EVENT(converter_currency_code, 
-                              smart_supply.amount / pow(10, currency.precision()), 
-                              reserve->contract, currency.code(), reserve_balance, ratio);
-    } else 
-        reserves_table.emplace(converter.owner, [&](auto& r) {
-            r.contract = contract;
-            r.ratio = ratio;
-            r.balance = asset(0, currency);
-        });
+    const auto reserve = reserves_table.find(currency.code().raw());
+    check(reserve == reserves_table.end(), "can't update existing reserve");
```

Comment:
maybe change the error to "reserve already exists" or something of that sort?

Reply:
Done :)